ui/pages/token-allowance/token-allowance.test.js

```diff
@@ -0,0 +1,332 @@
+import React from 'react';
+import configureMockStore from 'redux-mock-store';
+import { fireEvent } from '@testing-library/react';
+import { renderWithProvider } from '../../../test/lib/render-helpers';
+import { CHAIN_IDS } from '../../../shared/constants/network';
+import TokenAllowance from './token-allowance';
+
+const testTokenAddress = '0xC011a73ee8576Fb46F5E1c5751cA3B9Fe0af2a6F';
+const state = {
```

Comment:
I wonder if `mockState` from `../../../../test/data/mock-state.json` would have been sufficient here

Reply:
That will be useful to remove redundancy.

Reply:
I tried using mockState, but some properties are missing and the `gasEstimateType` property needs a different value. 
That is why I created a custom state in the test file.